### ISSUE TITLE
Automatic conversion from clojure types to equivalent XDMValues

### ISSUE THREAD

**Issue by davidmathei** [daveliepmann/uruk#8]
While I can pass on a variable value as, for instance, `{:group-plan-ids {:value [1 2 3] :type :sequence}}`, the error that gets returned tells me that I should convert the values first to XdmValues:
`Exception java.lang.IllegalArgumentException: Value must be array of XdmValue  com.marklogic.xcc.ValueFactory.newSequenceValue ...`

Would it be feasible to have uruk take care of that conversion?

**Comment by daveliepmann** [comment]
You're right, there should be some form of default conversion for Clojure
types into compatible Xdm types. There may be some conversion scenarios
that don't make sense. I'll take a look.

**Comment by davidmathei** [comment]
I experimented a bit with sequences of integers, but got stuck in xcc itself:
https://github.com/marklogic/xcc-java/blob/master/com/marklogic/xcc/impl/RequestImpl.java#L92

So setting of sequences is not even possible with XCC ...

**Comment by daveliepmann** [comment]
Confirmed:

```
(:import [com.marklogic.xcc ValueFactory]
         [com.marklogic.xcc.types XdmValue])

(with-open [sess (create-session db)]
  (execute-xquery sess "\"hello world!\""
                  {:variables {"test" {:value (into-array XdmValue
                                                          [(ValueFactory/newXSInteger 1)])
                                       :type :sequence}}}))
```

Other folks [who have encountered the same issue](http://developer.marklogic.com/pipermail/general/2010-March/004912.html) have worked around it by [passing XML documents that contain the desired sequence](http://developer.marklogic.com/pipermail/general/2010-March/004918.html).

I'll look at the issue a little more to see if other types will benefit from automatic conversion.

**Comment by daveliepmann** [comment]
@davidmathei Would an ArrayNode variable help you with your query?

```
(with-open [session (create-session db)]
  (execute-xquery session "xquery version \"1.0-ml\";
                                declare variable $my-variable as array-node() external;
                                $my-variable"
                  {:variables {"my-variable" {:value (ValueFactory/newArrayNode (str [1]))
                                              :type :array-node}}
                   :shape :single!}))
```

**Comment by davidmathei** [comment]
Meanwhile I resorted to pass all non-primitives as xml-node (which works fine). I will have a look at the ArrayNode when I run into simple sequences again.

**Comment by daveliepmann** [comment]
I just released version 0.2.5, which has automatic conversion of many variable types. This includes array-node, so you can just pass it something like `[1 2 3]`.

I'm closing this issue for now, since sequences are not supported and 0.2.5 addresses most common variable types.
